lib/detail_screen.dart

```diff
@@ -0,0 +1,288 @@
+import 'package:flutter/material.dart';
+import 'models.dart';
+import 'style.dart';
+import 'common_widgets.dart';
+import 'dart:math';
+import 'package:jiffy/jiffy.dart';
+
+class TagDetailPage extends StatefulWidget {
+  final String title;
+
+  const TagDetailPage({Key? key, required this.title}) : super(key: key);
+
+  @override
+  State<TagDetailPage> createState() => _TagDetailState();
+}
+
+class _TagDetailState extends State<TagDetailPage> {
+  late List<Expense> _expenses;
+  late Map<String, MonthwiseAggregatedExpense> _monthwiseAggregatedExpenses;
+  late ValueNotifier<MonthwiseAggregatedExpense> _showExpenseOfMonth;
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(() {
+      // TODO - get the height of each list view item post rendering
+      int itemHeight = 100;
+      double scrollOffset = _scrollController.offset;
```

Comment:
why do we need to get the height of list item?

Reply:
To be able to calculate which list item is on the top. The current value of 100 px is an approximation.